test/app.go

```diff
@@ -257,11 +244,16 @@ func (s *testSettings) apply() {
 	s.propertyLock.RUnlock()
 
 	for _, listener := range listeners {
-		select {
-		case listener <- s:
-		default:
-			l := listener
-			go func() { l <- s }()
-		}
+		listener <- s
 	}
+
+	s.app.propertyLock.Lock()
+	s.app.driver.CallFromGoroutine(func() {
+		painter.ClearFontCache()
+		cache.ResetThemeCaches()
+		intapp.ApplySettings(s, s.app)
+	})
+
+	s.app.appliedTheme = s.Theme()
+	s.app.propertyLock.Unlock()
```

Comment:
This locking around it all seems strange to me. Is it really safe? Should it not be in the scheduled function?

Reply:
Both needed the lock. I have split it in two - is that better?